schulze_condorcet/io.py

```diff
@@ -0,0 +1,66 @@
+"""Offer convenient functions to convert votes and candidates into proper types.
+
+This especially includes marking strings as Candidate or VoteString, and converting the
+string-based representations of votes (like 'a>b=c=d>e') into the level-based
+representations of votes (like [[a], [b, c, d], [e]]) and vice versa.
+"""
+
+from typing import Collection, List, Sequence, Union
+
+from schulze_condorcet.types import Candidate, VoteList, VoteString, VoteTuple
+
+
+def as_candidate(value: str) -> Candidate:
+    """Mark a string as candidate."""
+    return Candidate(value)
+
+
+def as_candidates(values: Sequence[str]) -> List[Candidate]:
+    """Mark a row of strings as candidates.
+
+    We respect the order of the candidates, as this is also respected during evaluation
+    of the votes using the schulze method.
+    """
+    return [Candidate(value) for value in values]
+
+
+def as_vote_string(
+        value: Union[str, VoteList, VoteTuple]
+) -> VoteString:
+    """Convert a level-based representation of the vote into its string representation.
+
+    This also accepts a string which will be marked as vote for convenience.
```

Comment:
This sounds like a bad idea. It basically negates the advantage of typing.

Reply:
Why do you think so? IMO the great advantage of typing is that you know what you give to and expect from function calls. Calling this function with a string is a conscious decision to convert it into a VoteString, similar as if you marked it as a VoteString manually by calling VoteString(mystr).

I just thought it would be nice to have one convenient method to mark / convert something into a VoteString. If you have a strong opinion about this, I am also fine with removing the string special case here.

Reply:
We already have a method for marking strings as VoteString namely VoteString(). Thus this introduces additional complexity to the function without clear upside.